lib/xebow/rgb_matrix/animation.ex

```diff
@@ -1,18 +1,27 @@
 defmodule Xebow.RGBMatrix.Animation do
   alias Xebow.RGBMatrix
 
-  @callback init_state(pixels :: list(RGBMatrix.pixel())) :: map
+  @callback init_state(pixels :: list(RGBMatrix.pixel())) :: t
   @callback next_state(
               pixels :: list(RGBMatrix.pixel()),
-              state :: map
-            ) :: map
+              animation :: t
+            ) :: t
+
+  @type t :: %__MODULE__{
+          tick: non_neg_integer,
+          speed: non_neg_integer,
+          delay_ms: non_neg_integer,
```

Comment:
The reason I hadn't done this is because "maybe all animations don't work like this".  The example animations I had originally provided are all of one class of animation, but the project I'm basing all this off of (QMK) has other classes of "animations", that are maybe better not described as animations at all...  We should talk more deeply about this soon, it's harder to do it all over test in github comments.

Reply:
Sure, happy to chat about it!

Reply:
Note for other readers: We had the conversation around this during the April 21 meeting. Some meeting notes are captured in the GitHub wiki.